tcms/management/management/commands/baseurl.py

```diff
@@ -0,0 +1,34 @@
+from django.core.management.base import BaseCommand
+from django.contrib.sites.models import Site
+
+
+class Command(BaseCommand):
+    help = "Sets the base URL of KiwiTCMS instance. " \
+           "If no arguments given returns current base URL."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'baseurl', nargs='?', default=None,
+            help='Base URL of KiwiTCMS instance',
+        )
+        parser.add_argument(
+            'basename', nargs='?', default='KiwiTCMS',
+            help='Optional site name. Defaults to "KiwiTCMS"',
+        )
+
+    def handle(self, *args, **kwargs):
+        if not kwargs['baseurl']:
+            try:
+                site = Site.objects.get(id=1)
+                self.stdout.write('URL: "%s", name: "%s"' % (site.domain, site.name))
+            except Site.DoesNotExist:
+                self.stdout.write('Base URL is not defined')
+            return
+        site, created = Site.objects.update_or_create(
```

Comment:
No fall back, this record always exists in the DB. If it doesn't then something is massively wrong.

Reply:
I found that if for some reason `settings.SITE_ID` is changed in common.py  the record with the new ID is not auto created and an exception is generated: django.contrib.sites.models.DoesNotExist: Site matching query does not exist.
This is highly unusual, but still we use this setting and update_or_create may be useful?

Reply:
IDK why that happens but it is not relevant. SITE_ID should always be 1 and the respective record should always be present so we don't care about the other scenarios.